File: engine/model/src/main/scala/tools/packet/parcel/SendDuey.scala
package tools.packet.parcel

import client.DueyAction
import net.opcodes.SendOpcode
import server.DueyPackage
import tools.packet.PacketInput

class SendDuey(private var _operation: DueyAction, private var _packages: Option[java.util.List[DueyPackage]]) extends PacketInput {
  def operation: DueyAction = _operation

  def packages: Option[java.util.List[DueyPackage]] = _packages

  def this(_operation: DueyAction) = {
    this(_operation, Option.empty)
  }

  override def opcode(): SendOpcode = SendOpcode.PARCEL
}

File: engine/engine-model/src/main/java/tools/packet/parcel/SendDuey.java
package tools.packet.parcel;

import java.util.ArrayList;
import java.util.List;

import client.DueyAction;
import net.opcodes.SendOpcode;
import server.DueyPackage;
import tools.packet.PacketInput;

public record SendDuey(DueyAction operation, List<DueyPackage> packages) implements PacketInput {
   public SendDuey(DueyAction operation) {
      this(operation, new ArrayList<>());
   }

   @Override
   public SendOpcode opcode() {
      return SendOpcode.PARCEL;
   }
}